File: src/aggregation/aggregation.py
"""
Aggregation Layer — Python Edition
"""

def aggregate_progress(events):
    return sum(e.delta for e in events)

def aggregate_health_scores(scores):
    return sum(s.value for s in scores) / len(scores)

def track_milestones(milestones):
    return len([m for m in milestones if m.completed])

File: src/aggregation/aggregation.js
/**
 * Aggregation Layer — JS Edition
 */

export function aggregateProgress(events) {
  return events.reduce((a, e) => a + e.delta, 0);
}

export function aggregateHealthScores(scores) {
  return scores.reduce((a, s) => a + s.value, 0) / scores.length;
}

export function trackMilestones(milestones) {
  return milestones.filter(m => m.completed).length;
}
